File: backend/api/src/user/song.py
from database import db


class Song(db.Model):
    series = db.Column(db.String(64))
    url = db.Column(db.String(64), unique=True)
    uri = db.Column(db.String(64), unique=True, primary_key=True)

    def __repr__(self):
        return "<The uri for {0} is {1}>".format(self.series, self.uri)

File: java-backend/src/main/java/com/anify/backend/model/Song.java
package com.anify.backend.model;

import jakarta.persistence.*;
import lombok.Data;
import com.fasterxml.jackson.annotation.JsonBackReference;

@Entity
@Data
public class Song {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String title;

    @Column(nullable = false)
    private String artist;

    @Column(name = "spotify_id")
    private String spotifyId;

    @Column(nullable = false)
    private String uri;

    @Column(nullable = false)
    private String url;

    @ManyToOne
    @JoinColumn(name = "series_id")
    @JsonBackReference
    private Series series;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    @JsonBackReference
    private User user;
} 